google/cloud/spanner_dbapi/connection.py

```diff
@@ -177,7 +217,7 @@ def _rerun_previous_statements(self):
         for statement in self._statements:
             res_iter, retried_checksum = self.run_statement(statement, retried=True)
             # executing all the completed statements
-            if statement != self._statements[-1]:
+            if statement != self._statements[-1] and not self.read_only:
```

Comment:
I think it would be better to change the `except Aborted` blocks in the different methods so that they never call `retry_transaction` when it is a read-only transaction. Instead the `Aborted` error should just be propagated to the caller when it is a read-only transaction (and in theory it should also never happen).